### DISCUSSION TITLE
Hello

### DISCUSSION THREAD

**Discussion by oliverwilms** [jgm/pandoc#10128] [General]
I am brand-new to pandoc. I try to programmatically create pdf from html source. I use command pandoc -o hello.pdf hello.html.

**Comment by jgm** [comment]
OK...but what is your question?  If you are having trouble, tell us exactly what it is.

**Reply by oliverwilms** [reply]
Hello, thank you for replying. I do not get the desired output pdf as expected. I realize I need to make the css and images available where pandoc is running. whatever is used in my html input file.

**Reply by oliverwilms** [reply]
How do I handle images and css in html file when I want to create a PDF?

**Reply by oliverwilms** [reply]
I just read about -c 

How about image to include in PDF file?

**Reply by oliverwilms** [reply]
I created a sample html for testing and also css. My command is pandoc -c /usr/irissys/csp/user/css/oliver.css -o /usr/irissys/csp/user/hello.pdf /usr/irissys/csp/user/hello.html

The output is:

Error producing PDF.
! Argument of \LT@nofcols has an extra }.
<inserted text> 
                \par 
l.87 \begin{longtable}[]{@{}lllll@{}}

**Reply by oliverwilms** [reply]
I have posted my test files here:
https://github.com/oliverwilms/pandoc-test/tree/main

**Reply by jgm** [reply]
`-c` option only affects HTML output.

Your input HTML is quite strange.  You have a table cell containing another table:
```
   <table border=0>
      <tr><td><table id="customers">
            <tr>
              <td>Customers</td>
```
I think pandoc isn't able to cope with that structure (nested tables) when producing LaTeX (or PDF via LaTeX).  You will get output if you specify that the PDF should be built from an HTML intermediary: `pandoc hello.html -o hello.pdf -t html -s`.  But, as I said above, if your aim is just to produce a PDF version of an HTML file, retaining its formatting, pandoc is not the right tool to use.

**Reply by oliverwilms** [reply]
I want to produce a PDF with formatting that looks like a table. I create HTML file to pass the data into table cells. How should I apply style to the PDF? I want to add a header with an image and background colors to certain table rows.

**Reply by jgm** [reply]
I can't do more here than repeat what I've already said.  Pandoc concerns itself with structure, not styling.  If you want to preserve CSS styling on tables, **do not use pandoc**.  Use one of several programs that will render an HTML to PDF the way the browser does, e.g. wkhtmltopdf or pagedjs-cli.

**Comment by jgm** [comment]
Pandoc will ignore CSS in the HTML input.  Images it should be able to handle.

If you just want to turn an HTML document into a PDF, preserving styling as in the browser, then you are much better off using a tool like pagedjs-cli.

**Reply by oliverwilms** [reply]
I am not having much CSS in my HTML input. I use many <table> <tr> <td> including table inside <td>.

Image relative path in HTML input will be picked up by pandoc? or should I specify image with absolute path?

**Reply by jgm** [reply]
relative path should work, if the image is actually there on the file system at that path.

**Reply by oliverwilms** [reply]
I feel I have invested significant effort to setup pandoc. I want to try to get it to work according to my needs. Appreciate your responses.
